compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1325,14 +1325,26 @@ static TileSizesListType getMmt4dTileSizes(linalg::LinalgOp op) {
 
   SmallVector<int64_t> distTileSizes =
       getDefaultDistributedLevelTileSizes(op, distConfig);
-  SmallVector<int64_t> parallelTileSizes(op.getNumLoops(), 1);
+  // Cache-level sizes are set to the distribution tile sizes for now. This will
+  // allow us to change distribution tile sizes while still preserving the
+  // existing cache behavior to some extent.
+  unsigned numLoops = op.getNumLoops();
+  SmallVector<int64_t> cacheParallelTileSizes(distTileSizes.begin(),
+                                              distTileSizes.end());
+  SmallVector<int64_t> cacheReductionTileSizes(numLoops, 0);
+
+  SmallVector<int64_t> parallelTileSizes(numLoops, 1);
   assert(parallelTileSizes.size() == mmt4dDimBase + 6);
   parallelTileSizes[mmt4dDimBase + 3] = M0;
   parallelTileSizes[mmt4dDimBase + 4] = N0;
   parallelTileSizes[mmt4dDimBase + 5] = K0;
   SmallVector<int64_t> reductionTileSizes;
   splitParallelAndReductionTiles(op, parallelTileSizes, reductionTileSizes);
-  return {distTileSizes, parallelTileSizes, reductionTileSizes};
+
+  SmallVector<int64_t> vectorInnerParallelTileSizes(numLoops, 0);
+  return {distTileSizes,           cacheParallelTileSizes,
+          cacheReductionTileSizes, parallelTileSizes,
```

Comment:
I'm curious if we really want to tile reduction loops in cache level? That will stop fusion, and introduce big stack allocation. It's fine for short-term, but we do plan to make mmt4d op fuse with its consumers. At that point, we could run into some issues.

Reply:
Mostly adding these levels to get it aligned with the rest of levels in TilingConfig and not have to add yet another version (hopefully we can start removing some of the versions there). I haven't experimented with the actual cache-level tile sizes yet but it's true that we are hitting some memory limitations. I understood that the use of local memory should fix any issue related to stock allocation, right? Do you have plans to enable it for the CPU backend?

Reply:
> I understood that the use of local memory should fix any issue related to stock allocation, right? 

Yes, I think so.

> Do you have plans to enable it for the CPU backend?

Not at this moment. I have other things on my plate.